src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/CreateFinancialTransactionCategoryIT.java

```diff
@@ -0,0 +1,180 @@
+        User testUser = createTestUser();
+        var categoryNameForWrongPathTest  = "`-'+=|\\/?,.<>%&(){}[];:" + "\"";
+        var financialTransactionCategoryCreateDTO = new FinancialTransactionCategoryCreateDTO(
+                categoryNameForWrongPathTest,
+                FinancialTransactionType.INCOME,
+                1L);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/categories")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(SecurityMockMvcRequestPostProcessors.user(String.valueOf(testUser.getId()))))
+                .andExpectAll(
+                        MockMvcResultMatchers.status().isBadRequest(),
+                        MockMvcResultMatchers.jsonPath("$.status")
+                                .value(ErrorCode.TEA003.getBusinessStatus()),
+                        MockMvcResultMatchers.jsonPath("$.message")
+                                .value(ErrorCode.TEA003.getBusinessMessage()),
+                        MockMvcResultMatchers.jsonPath("$.statusCode")
+                                .value(ErrorCode.TEA003.getBusinessStatusCode())
+                );
+    }
+
+    @DisplayName("Should return ResponseStatus BadRequest when type is empty")
+    @Test
+    void testCreateFinancialTransactionCategory_whenTypeIsEmpty_thenShouldReturnBadRequest() throws Exception {
+        User testUser = createTestUser();
+        var financialTransactionCategoryCreateDTO = new FinancialTransactionCategoryCreateDTO(
+                "Category",
+                null,
+                1L);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/categories")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(SecurityMockMvcRequestPostProcessors.user(String.valueOf(testUser.getId()))))
+                .andExpectAll(
+                        MockMvcResultMatchers.status().isBadRequest(),
+                        MockMvcResultMatchers.jsonPath("$.status")
+                                .value(ErrorCode.TEA003.getBusinessStatus()),
+                        MockMvcResultMatchers.jsonPath("$.message")
+                                .value(ErrorCode.TEA003.getBusinessMessage()),
+                        MockMvcResultMatchers.jsonPath("$.statusCode")
+                                .value(ErrorCode.TEA003.getBusinessStatusCode())
+                );
+    }
+
+    private User createTestUser() {
+        final User userOne = User.builder()
+                .id(1L)
+                .userName("UserOne")
+                .email("user@server.domain.com")
+                .password("Password1@")
+                .userStatus(UserStatus.VERIFIED)
+                .build();
+        return userRepository.save(userOne);
```

Comment:
Please add one empty line above the `return` statement for better code visibility.